lib/v2/screens/sign_up/display_name/display_name.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/i18n/edit_name.i18n.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/text_form_field_custom.dart';
+import 'package:seeds/v2/design/app_theme.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+
+class DisplayName extends StatefulWidget {
+  @override
+  _DisplayNameState createState() => _DisplayNameState();
+}
+
+class _DisplayNameState extends State<DisplayName> {
+  late SignupBloc _bloc;
+  final _keyController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    _bloc = BlocProvider.of<SignupBloc>(context);
+    _keyController.text = _bloc.state.displayNameState.displayName ?? '';
+    _keyController.addListener(() {
+      setState(() {
+        // Do nothing
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    _keyController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: _navigateBack,
+      child: Scaffold(
+        appBar: AppBar(),
+        body: BlocConsumer<SignupBloc, SignupState>(
+            listener: (context, state) {},
+            builder: (context, state) {
+              return Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: <Widget>[
+                    TextFormFieldCustom(
+                      labelText: 'Full Name'.i18n,
+                      onFieldSubmitted: (_) => _onNextPressed(),
+                      maxLength: 36,
+                      controller: _keyController,
+                      validator: (String? value) {
+                        if (value?.isEmpty == true) {
```

Comment:
didnt u create a function called isNullOrEmpty?

Reply:
yup, leftover from UI implementation. Fixed it